learn/what_is_meilisearch/comparison_to_alternatives.md

```diff
@@ -9,12 +9,144 @@
+| [Django](https://roadmap.meilisearch.com/c/60-django) | ❌ | ✅ | ❌ | ❌ |
+| [Rails](https://github.com/meilisearch/meilisearch-rails) | ✅ | ✅ | 🔶 <br>WIP | ❌ ||
+| [Official Laravel Scout Support](https://github.com/laravel/scout) | ✅ | ✅ | ❌ | ❌ |
+| [UI Search Kit](https://github.com/meilisearch/instant-meilisearch) | ✅ | ✅ | ✅ | ✅ |
+| [Docsearch](https://github.com/meilisearch/docs-scraper) | ✅ | ✅ | ✅ | ❌ |
+| [Strapi](https://github.com/meilisearch/strapi-plugin-meilisearch) | ✅ | ✅ | ❌ | ❌ |
+| [Gatsby](https://github.com/meilisearch/gatsby-plugin-meilisearch) | 🔶 <br> January 2022 | ✅ | ✅ | ❌ |
+| [Firebase](https://github.com/meilisearch/firestore-meilisearch) | 🔶 <br> January 2022 | ✅ | ✅ | ❌ |
+
+#### Configure
+
+##### Document schema
+
+| -       | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| Schemaless | ✅ | ✅ | 🔶 <br>Automatic schema detection is supported but needs to be specified. | ✅ |
+| Automatic document ID detection | ✅ | ❌ | ❌ | ❌ |
+| Native document formats | `Json`, `NDJson`, `CSV` | `Json` | `Json`, `NDJson` | `Json`, `NDJson`, `CSV` |
+
+##### Relevancy
+
+| - | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| Typo tolerant  | ✅ | ✅ | ✅ | 🔶 <br>Need to be specified by fuzzy queries. |
+| Orderable ranking rules | ✅ | ✅ | 🔶 <br>Tie-breaking order is limited by a unique scoring rule. | ❌|
+| Custom rules | ✅ | ✅ | 🔶 <br>Limited to one default sorting rule. | 🔶 <br>Function score query
+| Query field weights | ✅ | ✅ | ✅ | ✅ |
+| Synonyms | ✅ | ✅ | ✅ | ✅ |
+| Stop words | ✅ | ✅ | ❌ | ✅ |
+| Automatic language detection | ✅ | ✅ | ❌ | ❌ |
+| All language supports | ✅ | ✅ | ❌ <br> Only space separated | ✅ |
+
+##### Security
+
+| - | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| API Key Management | ✅ | ✅ | ✅ | ✅ |
+| Tenant tokens & multi-tenant indexes | **Q1 2022** <br> Configurable end-user hard filters per index. | ✅  <br> Hard filters are not configurable per index for an end-user tenant key.   | ✅ <br> Hard filters are not configurable per index for an end-user tenant key. | ✅ <br> Role based |
+
+##### Search
+
+| - | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| Placeholder search | ✅ | ✅ | ✅ | ✅ |
+| Multi-index search | Early 2022 | ✅ | ✅ | ✅ |
+| Exact phrase search | ✅ | ✅ | ❌ | ✅ |
+| Geo search |  ✅  | ✅ | ✅ | ✅ |
+| Sort by  |  ✅  | 🔶 <br> Limited to one `sort_by` rule per index. Index have to be duplicated for each sort field and sort order. | 🔶 <br> Do not support sort on string field. | ✅ |
+| Filtering |  ✅ <br>Support complex filter queries with an SQL-like syntax.  | 🔶 <br>Does not support `OR` operation across multiple fields. | 🔶 <br>Does not support `OR` operation across multiple fields. | ✅ |
+| Facetting |  ✅ | ✅ | ✅ | ✅ |
+| Facet value searches | ❌ | ✅ | ✅ | ❌ |
```

Comment:
What's the difference between faceting and facet value searches?

Reply:
@gmourier

Reply:
Facetting is distributing values of a field to allow filtering with UI components (range slider, checkbox, radio btn etc).

Facet value searches is having a search capability for inside a facet distribution. It translates into the UI by having a search bar for a facet distribution to search for a value.

Reply:
@guimachiavelli, @ferdi05

I did some research to try to find a clearer formulation. (I've deleted my previous comment 

The term `Facetting` could be renamed `Facet Search` since the user has the possibility to search documents by facet components. (Clicking on a checkbox to refine a category etc..)

The term `Facet values search` could be renamed to `Search for facet values`.

It is more accurate to represent it this way with an image 👨‍🎨

![Capture d’écran 2022-01-12 à 09 22 55](https://user-images.githubusercontent.com/3692335/149090438-16f40aac-c359-48e5-a1d7-6c8c848b1fed.png)

Reply:
Thanks @gmourier for the clarification. In [our doc](https://docs.meilisearch.com/reference/features/filtering_and_faceted_search.html#faceted-search), we use `Faceted search` and I think that it's slightly clearer than `Facet Search`.
I'm not super happy with `Search for facet values`, but I can't come up with something better. Do you know how it is called by Algolia and Typesense? Also, knowing that we get a red cross here, is it worth mentioning? I'd be ok to remove this row.

Reply:
Although @gmourier's suggestions are technically correct, I'm not sure the distinction will be immediately clear to users. `Faceted search` and `Search for facet values` are almost the same words arranged differently, so it feels like they are just saying the same thing twice if you're quickly going through this table.

Only thing I can come up with for facet value search is `Search through facet values`, but I'm not confident it's significantly clearer.

@ferdi05, I think the table would be clearer without facet value search, but I think it depends on how decisive this functionality is for our users. My hunch is that it isn't that important (a naïve string matching might be enough for this use-case), but I think you're better equipped to make the call.

Reply:
thanks for your feedback @guimachiavelli.
I'd say that @gmourier has a better view of this.

Reply:
@guimachiavelli I just discussed this with @gmourier and he agrees that we can remove the `Search for facet values` row.